File: LeetCode/Find Pivot Index.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        ps = [nums[0]] * len(nums)
        
        for i in range(1, len(nums)):
            ps[i] = ps[i-1] + nums[i]
        
        for i in range(len(nums)):
            if (ps[i] - nums[i] == ps[-1] - ps[i]): return i
            
        return -1
        

File: LeetCode/Find Pivot Index.cpp
class Solution {
 public:
  int pivotIndex(vector<int>& nums) {
    vector<int> ps(nums.size() + 1);
    ps[0] = 0;

    for (int i = 1; i <= nums.size(); i++) {
      ps[i] = ps[i - 1] + nums[i - 1];
    }

    for (int i = 1; i <= nums.size(); i++) {
      if (ps[i - 1] == ps.back() - ps[i]) return i - 1;
    }

    return -1;
  }
};